ui/js/models/resultsets_store.js

```diff
@@ -783,6 +783,13 @@ treeherder.factory('ThResultSetStore', [
         };
 
         var prependResultSets = function(repoName, data) {
+            // This request returns the treeherder git revision running on the server.
+            // If this differs from the version chosen during the UI page load, show a message.
+            if($rootScope.serverRev && $rootScope.serverRev !== data.meta.serverrev && _.isUndefined($rootScope.dismissedServerRev)) {
+                thNotify.send("Treeherder revision has changed, please reload the page", "warning", true);
+                $rootScope.dismissedServerRev = true;
+            }
```

Comment:
Could you maybe add an error banner to the top of the page? That way it would persist. You could also have a helpful link to actually reload the page. Something like what I did in #1874 would probably work well. (except just under the top banner)